libs/tools/barista/src/transform.ts

```diff
@@ -236,6 +237,65 @@ export const relativeUrlTransformer: BaPageTransformer = async (source) => {
   return transformed;
 };
 
+export const sectionTransformer: BaPageTransformer = async (source) => {
+  const transformed = { ...source };
+
+  transformed.content = runWithCheerio(source.content, ($) => {
+    $('h2, h3').each((_, element) => {
+      let sectionId = $(element).attr('id');
+      if (!sectionId) {
+        sectionId = $(element).text().replace(' ', '-');
+      }
+      // console.log(sectionId);
+      $(element.attribs).append(
+        $(element).attr('contentSection', ''),
+        $(element).attr('id', sectionId),
+      );
+    });
+  });
+
+  return transformed;
+};
+
+export const tableOfContentGenerator: BaPageTransformer = async (source) => {
+  const transformed = { ...source };
+  let toc: TableOfContents[] = [];
+  if (source.toc) {
+    // generate TOC and at to source
+    // Find Headlines and corresponding subheadlines (h2 > h3)
+    transformed.content = runWithCheerio(source.content, ($) => {
+      const headlines = $('h2, h3');
+      let currentH2 = -1;
+      // TODO: Id is lost somewhere? --------------!
+      headlines.each((_index, headline) => {
+        const headlineId = $(headline).attr('id');
+        const headlineText = $(headline).text();
+        if (headline.tagName === 'h2') {
```

Comment:
Is it always a h2 and a h3?

Reply:
Yap every page is structured that way.
Including h4s would explode the toc since h4s are used to section a subsection